File: client/src/types/EquipmentOffer.ts
export enum DeliveryType {
  SelfPickup = 'SelfPickup',
  DeliveryPaid = 'DeliveryPaid',
  FreeDelivery = 'FreeDelivery',
  DeliveryOnly = 'DeliveryOnly'
}

export interface EquipmentOfferCreateDto {
  ownerId: string;
  title: string;
  description?: string;
  price: number;
  location: string;
  condition: string;
  deliveryType: DeliveryType;
  contactNumber?: string;
}

export interface EquipmentOfferUpdateDto {
  title?: string;
  description?: string;
  price?: number;
  location?: string;
  condition?: string;
  deliveryType?: DeliveryType;
  contactNumber?: string;
  isAvailable?: boolean;
}

export interface EquipmentOfferPhotoDto {
  id: number;
  url: string;
  isMain: boolean;
}

export interface EquipmentOfferDto {
  id: number;
  ownerId: string;
  ownerName: string;
  ownerProfileImageUrl?: string;
  ownerMemberSince: string;
  title: string;
  description?: string;
  price: number;
  location: string;
  condition: string;
  deliveryType: DeliveryType;
  contactNumber?: string;
  isAvailable: boolean;
  createdAt: string;
  updatedAt?: string;
  photos: EquipmentOfferPhotoDto[];
}

File: API/Entites/EquipmentOffer.cs
using System;

namespace API.Entites;

public enum DeliveryType
{
    SelfPickup = 0,           // "Self Pickup Only"
    DeliveryPaid = 1,         // "Delivery Available (Additional Fee)"
    FreeDelivery = 2,         // "Free Delivery Included"
    DeliveryOnly = 3          // "Delivery Only (No Pickup)"
}

public class EquipmentOffer
{
    public int Id { get; set; } // Primary key

    // The user who created the offer
    public string OwnerId { get; set; } = default!;
    public AppUser Owner { get; set; } = default!;

    // Offer details
    public required string Title { get; set; }
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public required string Location { get; set; }
    public required string Condition { get; set; } // "New" or "Used"
    public DeliveryType DeliveryType { get; set; } = DeliveryType.SelfPickup;
    public required string ContactNumber { get; set; } // Required contact number

    // Status and metadata
    public bool IsAvailable { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Photos
    public ICollection<EquipmentOfferPhoto> Photos { get; set; } = new List<EquipmentOfferPhoto>();
}
